config/m4/knem.m4

```diff
@@ -7,22 +7,25 @@ AC_ARG_WITH([knem],
            [], [with_knem=guess])
 
 AS_IF([test "x$with_knem" != xno],
-      [AS_IF([test ! -d $with_knem], 
-             [
-              AC_MSG_NOTICE([KNEM path was not found, guessing ...])
-              with_knem=$(${PKG_CONFIG} --variable=prefix knem || find /opt/knem* -name knem_io.h |xargs dirname |sed -e s,/include,,g)
-              ],
-              [:])
-       AC_CHECK_HEADER([$with_knem/include/knem_io.h],
-                       [BASE_CFLAGS="$BASE_CFLAGS -I$with_knem/include"
-                        BASE_CPPFLAGS="$BASE_CPPFLAGS -I$with_knem/include"
-                        AC_DEFINE([HAVE_KNEM], [1], [Enable the use of KNEM])
-                        transports="${transports},knem"
-                        knem_happy="yes"],
-                       [AC_MSG_WARN([KNEM requested but not found])
-                        AC_DEFINE([HAVE_KNEM], [0], [Disable the use of KNEM])])],
-      [AC_MSG_WARN([KNEM was explicitly disabled])
-       AC_DEFINE([HAVE_KNEM], [0], [Disable the use of KNEM])]
+    [AS_IF([test "x$with_knem" == xguess -o "x$with_knem" == xyes -o "x$with_knem" == x],
+           [AC_MSG_NOTICE([KNEM path was not found, guessing ...])
+            ucx_check_knem_dir=$(${PKG_CONFIG} --variable=prefix knem || find /opt/knem* -name knem_io.h |xargs dirname |sed -e s,/include,,g)],
```

Comment:
maybe use this opportunity and fix this detection logic to use `pkg-config --cflags knem` to get the include dir for knem, and then use AC_CHECK_DECL for something from knem.h, instead of AC_CHECK_HEADER (to avoid the need of constructing the full path to knem_io.h)